### ISSUE TITLE
Attributes of TestAnalysis task

### ISSUE THREAD

**Issue by himyss** [evovch/ACCULINNA_analysis_template#1]
Хотелось бы узнать, можно ли оставить в атрибутах класса TestAnalysys только DetMessages.

Мысль в том, чтобы в методе TestAnalysis::Init(DetEventFull*)
я получал те самые DetMessages(), для всех станций всех детекторов.
Таким образом, глобальных переменных таска будет совсем немного. Вместо этих:

	Short_t fNsubElems;

	TGo4EventElement** fSubElements; //[fNsubElems]

	TGo4CompositeEvent* fRteleSubEl;
	TGo4CompositeEvent* fLteleSubEl;

	DetEventStation* fSubsubEl_CsI_R;
	DetEventStation* fSubsubEl_SQX_R;
	DetEventStation* fSubsubEl_SQY_R;
	DetEventStation* fSubsubEl_tCsI_R;
	DetEventStation* fSubsubEl_tSQX_R;
	DetEventStation* fSubsubEl_tSQY_R;

	DetEventStation* fSubsubEl_CsI_L;
	DetEventStation* fSubsubEl_SQX_L;
	DetEventStation* fSubsubEl_SQY_L;
	DetEventStation* fSubsubEl_tCsI_L;
	DetEventStation* fSubsubEl_tSQX_L;
	DetEventStation* fSubsubEl_tSQY_L;
	DetEventStation* fSubsubEl_tSQ300;
	DetEventStation* fSubsubEl_SQ300;

Будут только: 
        DefMessage** fSQX_L;
        и тд

На данный момент, я получаю объект
 DefMessage** fSQX_L;
уже в методе TestAnalysis::ProcessEvent(DetEventFull*). Получается, что для каждого события будут считываться эти строки: 

	TClonesArray* CsI_R_clar = fSubsubEl_CsI_R->GetDetMessages();
	TClonesArray* SQX_R_clar = fSubsubEl_SQX_R->GetDetMessages();
	TClonesArray* SQY_R_clar = fSubsubEl_SQY_R->GetDetMessages();
	TClonesArray* tCsI_R_clar = fSubsubEl_tCsI_R->GetDetMessages();
	TClonesArray* tSQX_R_clar = fSubsubEl_tSQX_R->GetDetMessages();
	TClonesArray* tSQY_R_clar = fSubsubEl_tSQY_R->GetDetMessages();

	TClonesArray* CsI_L_clar = fSubsubEl_CsI_L->GetDetMessages();
	TClonesArray* SQX_L_clar = fSubsubEl_SQX_L->GetDetMessages();
	TClonesArray* SQY_L_clar = fSubsubEl_SQY_L->GetDetMessages();
	TClonesArray* tCsI_L_clar = fSubsubEl_tCsI_L->GetDetMessages();
	TClonesArray* tSQX_L_clar = fSubsubEl_tSQX_L->GetDetMessages();
	TClonesArray* tSQY_L_clar = fSubsubEl_tSQY_L->GetDetMessages();
	TClonesArray* tSQ300_clar = fSubsubEl_tSQ300->GetDetMessages();
	TClonesArray* SQ300_clar = fSubsubEl_SQ300->GetDetMessages();

	DetMessage **SQY_R = new DetMessage*[SQY_R_clar->GetEntries()];
	for(Int_t i=0;i<SQY_R_clar->GetEntries();i++){
		SQY_R[i] = (DetMessage*)SQY_R_clar->At(i);
	}

При попытках перекинуть эти локальные переменные в глобальные у меня были ошибки компиляции, что то пока не разобрался. 

Также, когда мы с тобой добавили эти переменные в список атрибутов таска:

	TGo4CompositeEvent* fRteleSubEl;
	TGo4CompositeEvent* fLteleSubEl;

	DetEventStation* fSubsubEl_CsI_R;
	DetEventStation* fSubsubEl_SQX_R;
	DetEventStation* fSubsubEl_SQY_R;
	DetEventStation* fSubsubEl_tCsI_R;
	DetEventStation* fSubsubEl_tSQX_R;
	DetEventStation* fSubsubEl_tSQY_R;

	DetEventStation* fSubsubEl_CsI_L;
	DetEventStation* fSubsubEl_SQX_L;
	DetEventStation* fSubsubEl_SQY_L;
	DetEventStation* fSubsubEl_tCsI_L;
	DetEventStation* fSubsubEl_tSQX_L;
	DetEventStation* fSubsubEl_tSQY_L;
	DetEventStation* fSubsubEl_tSQ300;
	DetEventStation* fSubsubEl_SQ300;
Мы не добавили их инициализацию в конструктор и деструктор.

**Comment by evovch** [comment]
Да. Не вижу проблем, если ты будешь сразу вытаскивать TClonesArray и сохранять указатели на них как члены данных класса. По большому счёту, даже указатель на объект полного события не нужно передавать в метод ProcessEvent() потому что на самом деле объект то один, просто после каждого вызова inTree->GetEntry(iEvent) меняет его содержимое.

Инициализацию упустил. По хорошему нужно инициализировать нулевыми указателями nullptr, а где используешь делать проверку, что значение не nullptr.
